pkg/common/util/dial.go

```diff
@@ -0,0 +1,79 @@
+package util
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// BlockingDial is a helper method to dial the given address, using optional TLS credentials,
+// and blocking until the returned connection is ready. If the given credentials are nil, the
+// connection will be insecure (plain-text).
```

Comment:
> If the given credentials are nil, the connection will be insecure (plain-text).

This seems pretty dangerous. Since there is pretty much no scenario under which SPIRE should be making non-TLS connections, I feel we should make TLS mandatory.

Reply:
I agree :100: